lib/reporters/base.js

```diff
@@ -368,105 +354,124 @@ function pad (str, len) {
 }
 
 /**
- * Returns an inline diff between 2 strings with coloured ANSI output
+ * Converts an inline diff between two strings to coloured text representation.
  *
  * @api private
- * @param {Error} err with actual/expected
- * @return {string} Diff
+ * @param {Object} diff structured differences actual from expected
+ * @return {string} The diff for output to the console.
  */
-function inlineDiff (err) {
-  var msg = errorDiff(err);
+function inlineDiff (diff) {
+  var indent = '      ';
+  var lines = [''];
+  diff.forEach(function (change) {
+    function colorize (line) {
+      if (change.added) {
+        return color('diff added', line);
+      }
+      if (change.removed) {
+        return color('diff removed', line);
+      }
+      return line;
+    }
+
+    change.value.split('\n').forEach(function (line, i) {
+      line = colorize(line);
+      if (i === 0) {
+        line = lines.pop() + line;
+      }
+      lines.push(line);
+    });
+  });
 
-  // linenos
-  var lines = msg.split('\n');
   if (lines.length > 4) {
     var width = String(lines.length).length;
-    msg = lines.map(function (str, i) {
-      return pad(++i, width) + ' |' + ' ' + str;
-    }).join('\n');
+    lines = lines.map(function (line, i) { return indent + pad(i + 1, width) + ' | ' + line; });
+  } else {
+    lines = lines.map(function (line) { return indent + line; });
   }
+  lines.push('');
```

Comment:
I'm having trouble following this diff.  Why is this here?

Reply:
This is add empty line after diff, just as in [old code], but use more high-level abstraction, which allow write similar code in the `inlineDiff` and `unifiedDiff` functions

[old code]: https://github.com/mochajs/mocha/blob/master/lib/reporters/base.js#L396